File: src/server/gkill/api/req_res/get_skill_request.go
package req_res

type GetSkillRequest struct {
	SessionID  string `json:"session_id"`
	LocaleName string `json:"locale_name"`
	Name       string `json:"name"`
	// Path を省くと SKILL.md の全文とファイル一覧、指定するとそのファイルの中身を返す。
	Path string `json:"path"`
	// MaxBytes が正でファイルがそれより大きいときは中身を返さない（AI へ渡す量を抑えるため）。
	MaxBytes int64 `json:"max_bytes"`
}

File: src/client/classes/api/req_res/get-skill-request.ts
'use strict'

import { GkillAPIRequest } from '../gkill-api-request'

export class GetSkillRequest extends GkillAPIRequest {
    name: string
    // path を空にすると SKILL.md の全文とファイル一覧、指定するとそのファイルの中身が返る
    path: string
    // max_bytes は AI（MCP）が渡す量を抑えるためのもの。画面は 0（上限なし）で呼ぶ
    max_bytes: number

    constructor() {
        super()
        this.name = ""
        this.path = ""
        this.max_bytes = 0
    }
}
